iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractIdentifiable.java

```diff
@@ -60,17 +64,19 @@ public boolean hasProperty(String key) {
 
     @Override
     public String getProperty(String key) {
-        return properties.get(key);
+        Object val = properties.get(key);
+        return val != null ? val.toString() : null;
```

Comment:
Could the property key really be null? In getPropertyNames() it seems you assume that it’s never null.

Reply:
no the property key is indeed assumed to be null.
Though the value associated to the key may be null.

Reply:
I think @mathbagu just want you to add a null check on the key as it is not done inside properties.get().

Reply:
I think the properties class throw the exception (NullPointerException) so it should be ok like this.